### PR TITLE
Introduce startup-silent-period mechanism to avoid partial assignments

### PR DESCRIPTION
**Pull request by zuston**
### What changes were proposed in this pull request?
Introduce startup-silent-period mechanism to avoid partial assignments

### Why are the changes needed?

When changing some coordinator's conf and then restart, coordinator will accept client getAssignment request immediately, but it will serve for jobs request based on the partial registered shuffle-servers, which will make some jobs gotten not enough required shuffle-servers and then slow the running speed.

I think we should make coordinator wait for more than one shuffle-server heartbeat interval before serving for client. During out-of-service, requests from client will fallback to slave coordinator.

### Does this PR introduce _any_ user-facing change?

Yes

### How was this patch tested?

UTs

### CODE REVIEW

**Comment by zuston** [comment]
PTAL @jerqi

**Comment by jerqi** [comment]
Is `startup-silent-period` a good name? What's name of similar mechanism of Yarn?

**Comment by zuston** [comment]
> Is `startup-silent-period` a good name? What's name of similar mechanism of Yarn?

No such config in Yarn. `Safe mode` is for HDFS. I think it's not proper in Uniffle scenarios

**Comment by jerqi** [comment]
> > Is `startup-silent-period` a good name? What's name of similar mechanism of Yarn?
> 
> No such config in Yarn. `Safe mode` is for HDFS. I think it's not proper in Uniffle scenarios
> 
> Do u have any ideas?

It's ok for Uniffle. I don't have another good ideas.

**Comment by zuston** [comment]
> It's ok for Uniffle. I don't have another good ideas.

Got it. I will rename to safemode.

**Comment by jerqi** [comment]
> > It's ok for Uniffle. I don't have another good ideas.
> 
> Got it. I will rename to safemode.

I means that startup-silent-period is ok.

**Comment by zuston** [comment]
OK. Do u have any other advice? @jerqi

File: coordinator/src/main/java/org/apache/uniffle/coordinator/CoordinatorConf.java [hunk 1]
```diff
@@ -159,6 +159,18 @@ public class CoordinatorConf extends RssBaseConf {
           .enumType(AbstractAssignmentStrategy.HostAssignmentStrategy.class)
           .defaultValue(AbstractAssignmentStrategy.HostAssignmentStrategy.PREFER_DIFF)
           .withDescription("Strategy for selecting shuffle servers");
+  public static final ConfigOption<Boolean> COORDINATOR_START_SILENT_PERIOD_ENABLED = ConfigOptions
+      .key("rss.coordinator.startup-silent-period.enabled")
+      .booleanType()
+      .defaultValue(false)
+      .withDescription("Enable the startup-silent-period to reject the assignment requests "
```
Comment (jerqi) [inline coordinator/src/main/java/org/apache/uniffle/coordinator/CoordinatorConf.java:166]: Should we add more description to explain why we shouldn't use `true` as default value?

Comment (zuston) [reply coordinator/src/main/java/org/apache/uniffle/coordinator/CoordinatorConf.java:166]: Done

**Comment by zuston** [comment]
Updated.
1. Add the doc of these configs
2. Explain why make the startup-slient-period stop default.

**Review by jerqi — APPROVED** [review]
LGTM, thanks @zuston , wait for CI

**Comment by zuston** [comment]
```
Error:  Tests run: 6, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 17.998 s <<< FAILURE! - in org.apache.uniffle.test.CoordinatorGrpcTest
Error:  rpcMetricsTest  Time elapsed: 1.035 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <0.0> but was: <1.0>
```

CI failed. https://github.com/apache/incubator-uniffle/actions/runs/3216468616/jobs/5258387024 #244 

Rerun it.

**Comment by jerqi** [comment]
Unrelated flaky test failed, I will rerun the flaky test

**Pull request merged**